core/server/themes/loader.js

```diff
@@ -0,0 +1,39 @@
+var debug = require('debug')('ghost:themes:loader'),
+    config = require('../config'),
+    events = require('../events'),
+    read = require('./read'),
+    settingsApi = require('../api/settings'),
+    updateConfigAndCache,
+    loadThemes,
+    initThemes;
+
+updateConfigAndCache = function updateConfigAndCache(themes) {
+    debug('loading themes', themes);
+    config.set('paths:availableThemes', themes);
+    settingsApi.updateSettingsCache();
+};
+
+loadThemes = function loadThemes() {
+    return read
+        .all(config.getContentPath('themes'))
+        .then(updateConfigAndCache);
+};
+
+initThemes = function initThemes() {
+    debug('init themes', settingsApi.cache.get('activeTheme'));
+
+    // Register a listener for server-start to load all themes
+    events.on('server:start', function readAllThemesOnServerStart() {
```

Comment:
> If we move it onto being read on first request

What i meant is to only load all themes, which are not active, on the first GET themes request.
My thinking was that there is actually no need to load the themes (which are not active) at all. Only when the user navigates to themes in the admin panel. But we can also load them on bootstrap in background 👍